### PR TITLE
Updated vsphere-csi-driver to 2.4.1

### PR DESCRIPTION
**Pull request by MartinWeindel**
**How to categorize this PR?**

/area control-plane
/kind task
/platform vsphere

**What this PR does / why we need it**:
Updated vsphere-csi-driver dependencies to version 2.4.1.
As the issue [ControllerUnpublishVolume should not fail when the Volume is available](https://github.com/kubernetes-sigs/vsphere-csi-driver/issues/526) is still open, the `vsphere-csi-driver-controller` image is still based on an own fork (see https://github.com/MartinWeindel/vsphere-csi-driver).
Besides the credentials for csi-node on the shoot-cluster have been removed, as they are not needed anymore. 

**Which issue(s) this PR fixes**:
Fixes #

**Special notes for your reviewer**:

**Release note**:

```other operator
Updated vsphere-csi-driver dependencies to version 2.4.1.
```

### CODE REVIEW

File: charts/internal/machine-controller-manager/seed/templates/clusterrole.yaml [hunk 1]
```diff
@@ -19,3 +19,20 @@ rules:
...
+  - coordination.k8s.io
+  resources:
+  - leases
+  resourceNames:
+  - machine-controller
+  - machine-controller-manager
+  verbs:
+  - get
+  - watch
+  - update
```
Comment (acumino) [inline charts/internal/machine-controller-manager/seed/templates/clusterrole.yaml:38]: @MartinWeindel Should we keep it in separate PR as this PR is for csi-driver update and the changes here  is related to MCM leader election. WDYT?

Comment (MartinWeindel) [reply charts/internal/machine-controller-manager/seed/templates/clusterrole.yaml]: Done, see https://github.com/gardener/gardener-extension-provider-vsphere/pull/198

**Review by briantopping — APPROVED** [review]
/lgtm
